src/extensionManager.ts

```diff
@@ -53,42 +54,51 @@ export class ExtensionManager {
 
     const failDiagnostics = vscode.languages.createDiagnosticCollection(`Jest (${workspaceFolder.name})`)
 
-    this.extByWorkspace[workspaceFolder.name] = new JestExt(
-      this.context,
-      workspaceFolder,
-      jestWorkspace,
-      channel,
-      pluginSettings,
-      this.debugCodeLensProvider,
-      this.debugConfigurationProvider,
-      failDiagnostics,
-      instanceSettings
+    this.extByWorkspace.set(
+      workspaceFolder.name,
+      new JestExt(
+        this.context,
+        workspaceFolder,
+        jestWorkspace,
+        channel,
+        pluginSettings,
+        this.debugCodeLensProvider,
+        this.debugConfigurationProvider,
+        failDiagnostics,
+        instanceSettings
+      )
     )
   }
+  registerAll() {
+    vscode.workspace.workspaceFolders.forEach(this.register, this)
+  }
   unregister(workspaceFolder: vscode.WorkspaceFolder) {
     this.unregisterByName(workspaceFolder.name)
   }
   unregisterByName(name: string) {
-    const extension = this.extByWorkspace[name]
+    const extension = this.extByWorkspace.get(name)
     if (extension) {
       extension.deactivate()
-      delete this.extByWorkspace[name]
+      this.extByWorkspace.delete(name)
     }
   }
   unregisterAll() {
-    Object.keys(this.extByWorkspace).forEach(key => this.unregisterByName(key))
+    const keys = this.extByWorkspace.keys()
+    for (const key of keys) {
+      this.unregisterByName(key)
+    }
   }
   shouldStart(workspaceFolderName: string): boolean {
-    const { commonPluginSettings: { enabledWorkspaceFolders, disabledWorkspaceFolders } } = this
-    return (
-      // start if enabledWorkspaceFolders contains current workspace
-      enabledWorkspaceFolders.includes(workspaceFolderName) ||
-      // dont start if disabledWorkspaceFolders doesn't contain current workspace
-      !disabledWorkspaceFolders.includes(workspaceFolderName)
-    )
+    const { commonPluginSettings: { disabledWorkspaceFolders } } = this
+    if (this.extByWorkspace.has(workspaceFolderName)) {
+      return false
+    } else if (disabledWorkspaceFolders.includes(workspaceFolderName)) {
```

Comment:
no need for `else`,  you can also consider consolidating these 2 conditions into a single if statement, might be more readable...

Reply:
I did this with the same motivation :)

Reply:
what is the reason for `else`?

Reply:
Ah, missed that. No reason for `else`, i'll split it in two separate conditions.